plugins/tech-insights/src/api/types.ts

```diff
@@ -26,3 +29,27 @@ export type Check = {
   description: string;
   factIds: string[];
 };
+
+/**
+ * Represents a Fact defined on the TechInsights backend.
+ *
+ * @public
+ */
+export interface InsightFact {
+  [factId: string]: {
+    timestamp: string;
+    version: string;
+    facts: Record<
+      string,
+      | number
+      | string
+      | boolean
+      | DateTime
+      | number[]
+      | string[]
+      | boolean[]
+      | DateTime[]
+      | JsonValue
```

Comment:
`JsonValue` covers basically every type except `undefined`. So `number`, `string`, `boolean`,`null`, arrays, and objects. Is that really intended here?

Reply:
I copied from https://github.com/backstage/backstage/blob/6c027cdab269b80227effbcbb3d2305f602490a3/plugins/tech-insights-node/src/facts.ts#L50 but makes keep just JsonValue sense